MapboxDirections/MBVisualInstructionComponent.swift

```diff
@@ -84,14 +88,57 @@ open class VisualInstructionComponent: NSObject, NSSecureCoding {
         }
         self.imageURL = imageURL
         
-        self.isDelimiter = decoder.decodeBool(forKey: "isDelimiter")
+        guard let typeString = decoder.decodeObject(of: NSString.self, forKey: "type") as String?, let type = VisualCompomentType(description: typeString) else {
+                return nil
+        }
+        self.type = type
     }
     
     open static var supportsSecureCoding = true
     
     public func encode(with coder: NSCoder) {
         coder.encode(text, forKey: "text")
         coder.encode(imageURL, forKey: "imageURL")
-        coder.encode(isDelimiter, forKey: "isDelimiter")
+        coder.encode(type, forKey: "type")
+    }
+}
+
+
+/**
+ `VisualCompomentType` describes the `text` type.
+ */
+@objc(MBVisualComponentType)
+public enum VisualCompomentType: Int, CustomStringConvertible {
```

Comment:
This enumeration is a little difficult to find here. Consider moving it to the top of the file, where enumerations and other non-class types are typically declared, or to a separate file.